src/qibotn/eval_qu.py

```diff
@@ -43,15 +10,24 @@ def init_state_tn(nqubits, init_state_sv):
     return qtn.tensor_1d.MatrixProductState.from_dense(init_state_sv, dims)
 
 
-def dense_vector_tn_qu(qasm: str, initial_state, is_mps, backend="numpy"):
+def dense_vector_tn_qu(qasm: str, initial_state, mps_opts, backend="numpy"):
     """Evaluate QASM with Quimb.
 
     backend (quimb): numpy, cupy, jax. Passed to ``opt_einsum``.
     """
-    circuit = QiboCircuit.from_qasm(qasm)
+
     if initial_state is not None:
-        initial_state = init_state_tn(circuit.nqubits, initial_state)
-    circ_quimb = from_qibo(circuit, is_mps, psi0=initial_state)
+        nqubits = int(np.log2(len(initial_state)))
+        initial_state = init_state_tn(nqubits, initial_state)
+
+    if mps_opts:
+        circ_quimb = qtn.circuit.CircuitMPS.from_openqasm2_str(
+            qasm, psi0=initial_state, gate_opts=mps_opts
+        )
+
+    else:
+        circ_quimb = qtn.circuit.Circuit.from_openqasm2_str(qasm, psi0=initial_state)
```

Comment:
```suggestion
    circ_quimb = qtn.circuit.CircuitMPS.from_openqasm2_str(
        qasm, psi0=initial_state, gate_opts=mps_opts
    )
```
Assume that `mps_opts` is always suitable for Quimb. Otherwise, set the default in a compatible way (i.e. `mps_opts=None` in the parameters).

Reply:
Circuit is for general 1d tensornetwork whereas CircuitMPS is specifically targeted for MPS. If not using Circuit means we don't need the condition MPS_enabled or not. For testing, we are using MPS so both can be applied. But, Circuit and CircuitMPS is using two different contraction stratergies. In CircuitMPS, MPS structure is kept whereas in the other it is not. I am afraid Circuit may be needed later while using MPO.

Reply:
I also wanted to avoid branching. One way is to use two different functions (one for mps and another general) as Andy is doing for cuquantum backend. Though not sure it is optimal one.

Reply:
```suggestion
    circ_cls = qtn.circuit.CircuitMPS if mps_opts else qtn.circuit.Circuit
    circ_quimb = circ_cls.from_openqasm2_str(qasm, psi0=initial_state, gate_opts=mps_opts)
```

My bad, I did not realize they were two different classes :)

Reply:
Updated accordingly.